Feature: Implement automated rating request

wutschel opened the pull request:
## Description
<!--- Detailed info for reviewers and developers -->
Closes https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/370.

This PR implements automated rating request for the AppStore. 

1st attempt: It will show the rating popup each 10th start/wakeup.

## Summary for release notes
<!--- This will be shown in Testflight to end users / testers -->
<!--- Please fill in, as usually PR title isn't clear to ordinary users -->
<!--- If your changes don't modify app files, please add prefix [not app] -->
Feature: Implement automated rating request

wutschel commented:
I added some logic which will save the current version in the key `"VersionUnderReview"` each time the version is updated. If the current version is different to the one under review, the counter will be reset to 0. If the version keeps same, the counter in key `"PlaybackAttempts"` is increased and each 50th time the review request will pop up. I am not 100% sure after how many attempts the popup shall appear. I am also thinking of adding a check for a certain amount of time to be passed before we ask again (e.g. 6 months or so). 

What are your thoughts?

wutschel commented:
@kambala-decapitator, do you see this is a good choice to trigger the reviews? And if so, should we add this for the next release already?

kambala-decapitator commented on XBMC Remote/Utilities.m:
```diff
@@ -859,13 +859,25 @@ + (void)showReviewController {
-    // Show review popup each 10th launch/wakeup
-    if (count % 10 == 0) {
-        [Utilities showReviewController];
+    id infoDict = NSBundle.mainBundle.infoDictionary;
```
why `id`? The type is known. Please either use `__auto_type` or `NSDictionary*`.

wutschel replied:
Will use NSDictionary.

kambala-decapitator commented on XBMC Remote/Utilities.m:
```diff
@@ -859,13 +859,25 @@ + (void)showReviewController {
+        NSInteger count = [[NSUserDefaults standardUserDefaults] integerForKey:@"PlaybackAttempts"] + 1;
+        [[NSUserDefaults standardUserDefaults] setInteger:count forKey:@"PlaybackAttempts"];
+        
+        // Show review popup each 10th launch/wakeup
+        if (count % 50 == 0) {
```
either code or comment is incorrect

wutschel replied:
Corrected the comment.

kambala-decapitator commented on XBMC Remote/Utilities.m:
```diff
@@ -859,13 +859,25 @@ + (void)showReviewController {
+    // Compare current version with version under review
+    if (![savedVersion isEqualToString:currentVersion]) {
+        // Reset counter to 0 for new version
+        [[NSUserDefaults standardUserDefaults] setObject:currentVersion forKey:@"VersionUnderReview"];
+        [[NSUserDefaults standardUserDefaults] setInteger:0 forKey:@"PlaybackAttempts"];
```
please don't use "magic" strings

wutschel replied:
Used defines now.

kambala-decapitator commented:
> @kambala-decapitator, do you see this is a good choice to trigger the reviews? And if so, should we add this for the next release already?

let's do it!

wutschel commented:
Included one more fix (comment still referred to "50th launch/wakeup", but the trigger comes after "50th playback attempt"). Squashed all changes into one and updated the commit message.

kambala-decapitator commented on XBMC Remote/Utilities.h:
```diff
@@ -88,5 +88,7 @@ typedef enum {
 + (BOOL)hasRemoteToolBar;
 + (CGFloat)getBottomPadding;
 + (void)sendXbmcHttp:(NSString*)command;
++ (void)showReviewController;
```
not sure why this method is public

wutschel replied:
A remnant of the first implementations I did. I now removed it as part of the latest forced push.

wutschel commented:
Updated with last request for change.
